mdbook-tera-backend/src/tera_renderer/custom_component.rs

```diff
@@ -0,0 +1,38 @@
+use anyhow::Result;
+use serde::Deserialize;
+use std::path::{Path, PathBuf};
+use tera::Tera;
+
+/// Configuration in `book.toml` `[output.tera-renderer]`.
+#[derive(Deserialize)]
+pub struct TeraRendererConfig {
+    /// Relative path to the templates directory from the `book.toml` directory.
+    pub templates_dir: PathBuf,
+}
+
+impl TeraRendererConfig {
+    /// Recursively add all templates in the `templates_dir` to the `tera_template`.
+    fn add_templates_recursively(tera_template: &mut Tera, directory: &Path) -> Result<()> {
```

Comment:
This should just be a regular function, no need to make it an associated function.